plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -241,163 +246,57 @@ void deleteBlobsByPrefix(String prefix) throws IOException {
-                                }
-
-                                @Override
-                                public void onSuccess(Void aVoid, HttpHeaders responseHeaders) throws IOException {
-                                    countDown.countDown();
-                                }
-                            });
-                        }
-
-                        batch.execute();
-
-                        if (countDown.isCountedDown() == false) {
-                            throw new IOException("Failed to delete all [" + deletions.size() + "] blobs");
-                        }
-                    } finally {
-                        deletions.clear();
-                    }
-                }
+        final List<BlobId> blobIdsToDelete = blobNames.stream().map(blobName -> BlobId.of(bucket, blobName)).collect(Collectors.toList());
+        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> storage.delete(blobIdsToDelete));
+        assert blobIdsToDelete.size() == deletedStatuses.size();
+        boolean failed = false;
+        for (int i = 0; i < blobIdsToDelete.size(); i++) {
+            if (deletedStatuses.get(i) == false) {
+                logger.error("Failed to delete blob [{}] in bucket [{}].", blobIdsToDelete.get(i).getName(), bucket);
+                failed = true;
             }
-        });
+        }
+        if (failed) {
+            throw new IOException("Failed to delete all [" + blobIdsToDelete.size() + "] blobs.");
+        }
     }
 
     /**
      * Moves a blob within the same bucket
      *
      * @param sourceBlob name of the blob to move
-     * @param targetBlob new name of the blob in the target bucket
+     * @param targetBlob new name of the blob in the same bucket
      */
-    void moveBlob(String sourceBlob, String targetBlob) throws IOException {
-        SocketAccess.doPrivilegedIOException(() -> {
+    void moveBlob(String sourceBlobName, String targetBlobName) throws IOException {
+        final BlobId sourceBlobId = BlobId.of(bucket, sourceBlobName);
+        final BlobId targetBlobId = BlobId.of(bucket, targetBlobName);
+        final CopyRequest request = CopyRequest.newBuilder()
+                .setSource(sourceBlobId)
+                .setTarget(targetBlobId)
+                .build();
+        SocketAccess.doPrivilegedVoidIOException(() -> {
             // There's no atomic "move" in GCS so we need to copy and delete
-            client.objects().copy(bucket, sourceBlob, bucket, targetBlob, null).execute();
-            client.objects().delete(bucket, sourceBlob).execute();
-            return null;
+            final CopyWriter copyWriter = storage.copy(request);
+            copyWriter.getResult();
+            final boolean deleted = storage.delete(sourceBlobId);
+            if (deleted == false) {
+                throw new IOException("Failed to move source [" + sourceBlobName + "] to target [" + targetBlobName + "].");
```

Comment:
Nit: Exception messages and log traces usually have no dot at the end, except if they contain more than 1 sentence.

Reply:
I haven't noticed this, thanks.